src/diff-snapshot.js

```diff
@@ -181,6 +176,27 @@ function diffImageToSnapshot(options) {
   return result;
 }
 
+
+function runDiffImageToSnapshot(options) {
+  options.receivedImageBuffer = options.receivedImageBuffer.toString('base64');
+
+  const serializedInput = JSON.stringify(options);
+
+  let result = {};
+  const writeDiffProcess = childProcess.spawnSync('node', [`${__dirname}/write-result-diff-image.js`], { input: Buffer.from(serializedInput) });
```

Comment:
I've updated the PR to use a fourth file descriptor for the return data.

The process can now write to stdout/stderr per usual, and we can get the results back without needing to introduce callbacks.

This may also address visibility into errors adequately.